azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java

```diff
@@ -1092,11 +1092,12 @@ public void testKillBeforeStart() throws Exception {
       }
     });
 
-    this.runner.run();
+    FlowRunnerTestUtil.startThread(this.runner).join();
     // children jobs shouldn't start
     assertStatus("joba", Status.READY);
     assertStatus("joba1", Status.READY);
     waitForAndAssertFlowStatus(Status.KILLED);
+    this.runner = null;
```

Comment:
Clarification: why are we doing this? On a different note, the base class has runner instance variable that is shared by many tests. what is the reason for having this as an instance variable rather than creating it for every unit test?

Reply:
Setting to _null_ to prevent other tests from using this particular instance.
I don't know the reason why _runner_ is an instance variable to be honest but that's something we should look into because as you mentioned tests should run in isolation from each other.